Tests/Integration/Transacting/MobileCoinClientInternalIntTests.swift

```diff
@@ -287,4 +287,173 @@ class MobileCoinClientInternalIntTests: XCTestCase {
+            XCTAssertEqual(clientBals.tokenIds, Set<TokenId>(try XCTUnwrap(balances.keys)))
+                        
+            for tokenId in clientBals.tokenIds {
+                let expectedAmt = try XCTUnwrap(balances[tokenId]).reduce(0, +)
+                let clientBal = try XCTUnwrap(clientBals.balances[tokenId])
+                let clientAmt = try XCTUnwrap(clientBal.amount())
+                XCTAssertEqual(clientAmt, expectedAmt)
+                
+                while true {
+                    let amtTransferable = try await client.amountTransferable(tokenId: tokenId)
+                    guard amtTransferable > 0 else {
+                        break
+                    }
+
+                    let returnAmt = Amount(amtTransferable, in: tokenId)
+                    let fee = try await client.estimateTotalFee(
+                        toSendAmount: returnAmt,
+                        feeLevel: .minimum)
+
+                    let transaction = try await client.prepareTransaction(
+                        to: returnAddress,
+                        amount: returnAmt,
+                        fee: fee)
+                    try await client.submitTransaction(transaction: transaction.transaction)
+                    
+                    var txComplete = false
+                    while !txComplete {
+                        try await client.updateBalances()
+                        let txStatus = try await client.status(of: transaction.transaction)
+                        switch txStatus {
+                        case .accepted(block: _ ):
+                            txComplete = true
+                        case .failed:
+                            print("Failed to return funds")
+                            break
+                        case .unknown:
+                            // throttle the polling a bit
+                            sleep(1)
+                        }
+                    }
+                    
+                    try await client.updateBalances()
+                }
+            }
+        }
+
+        let sourceAccountKey = try IntegrationTestFixtures.createAccountKey(accountIndex: 0)
+        let sourceClient = try await IntegrationTestFixtures.createMobileCoinClientWithBalance(
+            accountIndex: 0,
+            transportProtocol: .http)
+
+        let accountFactory = TestAccountFactory(
+            fogReportUrl:  NetworkConfigFixtures.network.fogReportUrl,
+            fogAuthoritySpki: try NetworkConfigFixtures.network.fogAuthoritySpki())
+
+        let testAccountConfigs = [
+            TestAccountFactory.TestAccountConfig(
+                name:"acct0",
+                txData:
+                    [.MOB : [1 + minFee]]),
+            TestAccountFactory.TestAccountConfig(
+                name:"acct1",
+                txData:
+                    [
+                        .MOB : [1, 2, minFee],
+                        .MOBUSD : [1 , 2, 3, minMOBUSDFee]
```

Comment:
if you like the convenience of this initializer w/ integers in the array, you could write a custom init function that converts those UInt64's into amounts, then we get the semantics of "Amount" without the annoyance of writing them each time.

Reply:
done